### PR TITLE
Migrate Zowe doc site generator

### PR DESCRIPTION
**Pull request by nannanli**
## New Features! 💥

This PR migrates the doc site static site generator from Vuepress to Docusaurus. This includes all public Zowe doc versions. 

- More depth to sidebars (supports nested topics)
-  Three column layout (nav-topic-mini toc) 
-  Collapsible sidebar (can be changed into two column layout as per need)
-  Two themes (dark and light)
-  Zoom-in option for small Doc Images
- Support reading docs offline
-  Social module
   - "Share" button that allows users to share a topic to social platforms
   -  "Discussion" button that opens Zowe Slack channels. It's in the footer section now as a resource. 
-  Feedback module
   -  Rate "Is this topic helpful" - thumbs up and thumbs down. Helpful or not helpful. When not helpful, users are asked to create a Github issue. 
   -  "Open doc issue" option for each topic that directs users to the doc site repo with associated topic URL and background information in the issue automatically.  
   -  Better visible "Edit this topic" link that directs users to the source file for each topic
- Can print PDF (download the complete doc into PDF
- Similar look and feel with zowe.org (includes footer) 
- Enhanced doc site UX with improved home page design

### Your checklist for this pull request

- [x] If the changes in this PR is part of the next future release, make this pull request against the **docs-staging** branch which will be published at the next release boundary. If the changes in this PR are part of the current release, use the default base branch, **master**. For more information about branches, see https://github.com/zowe/docs-site/tree/master#understanding-the-doc-branches. 
      
- [ ] If this PR relates to GitHub issues in `docs-site` or other repositories, please list in Description, prefixed with **close**, **fix** or **resolve** keywords.
Fixes: #1721 #1626 #1587

### Related issues

Some existing issues might be able to be addressed in the process. 

- #1376 Add 'share to social media' buttons on Zowe doc site 
- #1259 Doc Enhancement: Track and alert on last-updated topics 
- #730 Enhancement - Add Slack button, other UI improvements
- #558 Inconsistent Bold Headings in TOC
- #533 Enhancement - Dark Theme option
- #286 Allow Users to adjust the size of the TOC frame in the browser
- #653 Add Home Button/Icon to the Toolbar/nav menus
- #1672 Large Images are small in size and not properly readable

### CODE REVIEW

File: docs/extend/extend-apiml/api-mediation-caching-service.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+# Using the Caching Service
```
Comment (covalentbond) [inline docs/extend/extend-apiml/api-mediation-caching-service.md:1]: @nannanli this `extend-apiml/api-mediation-catching-service` & `extend-apiml/api-mediation-redis` needs to be referenced in the [sidebar.js](https://github.com/zowe/docs-site/blob/zowe-docs-v2/sidebars.js#L178).

Comment (covalentbond) [inline docs/extend/extend-apiml/api-mediation-caching-service.md:1]: @nannanli these new files `extend-apiml/api-mediation-catching-service` & `extend-apiml/api-mediation-redis` added in this [commit](https://github.com/zowe/docs-site/pull/1725/commits/96bd7fb1c7f0763f2f91344b86444c74f47a9f82) needs to be referenced in the [sidebar.js](https://github.com/zowe/docs-site/blob/zowe-docs-v2/sidebars.js#L178)

Comment (covalentbond) [inline docs/extend/extend-apiml/api-mediation-caching-service.md:1]: @nannanli these following files added in this [commit](https://github.com/zowe/docs-site/pull/1725/commits/96bd7fb1c7f0763f2f91344b86444c74f47a9f82) needs to be referenced in the [sidebar.js](https://github.com/zowe/docs-site/blob/zowe-docs-v2/sidebars.js#L178):

- `extend-apiml/api-mediation-catching-service`
- `extend-apiml/api-mediation-redis`
- `api-mediation-vsam.md`

Comment (covalentbond) [inline docs/extend/extend-apiml/api-mediation-caching-service.md:1]: @nannanli these following files added in this [commit](https://github.com/zowe/docs-site/pull/1725/commits/96bd7fb1c7f0763f2f91344b86444c74f47a9f82) needs to be referenced in the [sidebar.js](https://github.com/zowe/docs-site/blob/zowe-docs-v2/sidebars.js#L178):

- `extend-apiml/api-mediation-catching-service`
- `extend-apiml/api-mediation-redis`
- `extend-apiml/api-mediation-vsam.md`

Comment (nannanli) [reply docs/extend/extend-apiml/api-mediation-caching-service.md:1]: Done. : )

File: versionsArchived.json [hunk 2]
```diff
@@ -1,4 +1,7 @@
 {
+  "v1.11.x": "https://docs.zowe.org/v1-11-x/",
+  "v1.10.x": "https://docs.zowe.org/v1-10-x/",
+  "v1.9.x": "https://docs.zowe.org/v1-9-x/",
```
Comment (covalentbond) [inline versionsArchived.json]: @nannanli once this PR is merged. I don't think the vuepress links will be working anymore.

We will either need to make standalone Docusaurus deploy links for every version in different branches. Or for now we can redirect to links as done in this [versionsArchived.json](https://github.com/covalentbond/docs-site/blob/zowe-docs-v2/versionsArchived.json)

**Pull request closed without merging**